src/txmempool.cpp

```diff
@@ -247,6 +247,24 @@ void CTxMemPool::removeConflicts(const CTransaction &tx, std::list<CTransaction>
     }
 }
 
+void CTxMemPool::removeExpired(unsigned int nBlockHeight)
+{
+    // Remove expired txs from the mempool
+    LOCK(cs);
```

Comment:
Is this lock sufficient to protect the map from other threads?  Hmm! cc @str4d @daira

Reply:
Yes. `cs` is specifically the lock for `mapTx` (which I read from `cs` being directly adjacent to `mapTx` in `txmempool.h`).